packages/react-core/src/layouts/Grid/Grid.js

```diff
@@ -3,31 +3,63 @@ import PropTypes from 'prop-types';
 import styles from '@patternfly/patternfly-next/layouts/Grid/styles.css';
 import { css } from '@patternfly/react-styles';
 import { GutterSize, getGutterModifier } from '../../styles/gutters';
+import { getGridSpanModifier, gridSpans } from './gridUtils';
+import { DeviceSizes } from '../../styles/sizes';
+
+const gridItemSpanValueShape = PropTypes.oneOf(gridSpans);
 
 const propTypes = {
   children: PropTypes.any,
   className: PropTypes.string,
-  gutter: PropTypes.oneOf(Object.keys(GutterSize))
+  gutter: PropTypes.oneOf(Object.keys(GutterSize)),
+  span: gridItemSpanValueShape,
+  xs: gridItemSpanValueShape,
+  sm: gridItemSpanValueShape,
+  md: gridItemSpanValueShape,
+  lg: gridItemSpanValueShape,
+  xl: gridItemSpanValueShape
 };
 
 const defaultProps = {
   children: null,
   className: '',
-  gutter: null
+  gutter: null,
+  span: null,
+  xs: null,
+  sm: null,
+  md: null,
+  lg: null,
+  xl: null
 };
 
-const Grid = ({ children, className, gutter, ...props }) => (
-  <div
-    className={css(
-      styles.grid,
-      gutter && getGutterModifier(styles, gutter, styles.modifiers.gutter),
-      className
-    )}
-    {...props}
-  >
-    {children}
-  </div>
-);
+const Grid = ({ children, className, gutter, span, ...props }) => {
+  const classes = [styles.grid, span && getGridSpanModifier(span)];
+
+  Object.keys(DeviceSizes).forEach(size => {
```

Comment:
Good choice to duplicate this here.  I think if we moved this and the GridItem one into a shared area it would be too complicated to follow.